astropy/units/physical.py

```diff
@@ -328,7 +328,7 @@ class PhysicalType:
 
     def __init__(self, unit, physical_types):
         self._unit = _replace_temperatures_with_kelvin(unit)
-        self._physical_type_id = self._unit._get_physical_type_id()
+        self._id = self._unit._physical_type_id
```

Comment:
I'm not sure there is a need for this rename - it really is the same thing with the same meaning. 

That said, now that the unit's item is cached, I think we could also just remove this attribute and use `._unit._physical_type_id`... Given how little the attribute is used, maybe that is best.

Reply:
`PhysicalType` cannot have an attribute called `_physical_type_id` or the code will think it is a unit because of a change in `astropy/units/core.py`:
```diff
          # Short-circuit if we're already a unit
-         if hasattr(s, "_get_physical_type_id"):
+         if hasattr(s, "_physical_type_id"):
              return s
```
But I'll remove the attribute instead of renaming it.

Reply:
Ah, yes, that's a good point! The hack I introduced coming back to bite - thanks for catching that! We really need an abstract type for units...